### PR TITLE
Forbid #require in dune files in OCaml syntax

### PR DESCRIPTION
**Pull request by diml**
`#require` was restricted to `#require "unix"` a while ago as allowing files in OCaml syntax to load anything didn't seem right. We initially kept `"unix"` as some projects were using it, but we can now forbid `#require` entirely in `dune` files.

Forbidding `unix` improves portability.

### CODE REVIEW

**Comment by diml** [comment]
I looked at the projects using `#require "unix"` in the dune-universe. It's mostly used for reading the output of a command via `Unix.open_process_in ...`.  It seems a bit painful to do without unix, so I added a function `run_and_read_lines` in the plugin API.

While we don't want to encourage the use of the OCaml syntax, we don't support enough yet to replace the current uses cases for it, so it doesn't seem worth making it difficult for users to switch from jbuilder to dune just because of not allowing `unix` anymore.

File: src/jbuild_load.ml [hunk 1]
```diff
@@ -106,6 +118,29 @@ module Jbuild_plugin = struct
...
+    let run_and_read_lines cmd =
+      let tmp_fname = Filename.temp_file "dune" ".output" in
+      at_exit (fun () -> Sys.remove tmp_fname);
+      let n =
+        Printf.ksprintf Sys.command "%%s > %%s" cmd (Filename.quote tmp_fname)
+      in
+      let rec loop ic acc =
+        match input_line ic with
+        | exception End_of_file -> close_in ic; List.rev acc
+        | line -> loop ic (line :: acc)
```
Comment (rgrinberg) [inline src/jbuild_load.ml]: Can we move this to the `Io` module? Note that I see so much opportunity for code reuse, but keeping the application code uncluttered is nice.

Comment (diml) [reply src/jbuild_load.ml]: It's not a trivial move since this is for plugins

Comment (rgrinberg) [reply src/jbuild_load.ml]: Ah yeah, that's right. Bit of a shame.

**Pull request merged**